iceoryx_posh/test/moduletests/test_popo_server_options.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/popo/server_options.hpp"
+
+#include "test.hpp"
+
+namespace
+{
+using namespace ::testing;
+
+TEST(ServerOptions_test, SerializationRoundTripIsSuccessful)
+{
+    iox::popo::ServerOptions defaultOptions;
+    iox::popo::ServerOptions testOptions;
+
+    testOptions.requestQueueCapacity = 42;
+    testOptions.nodeName = "hypnotoad";
+    testOptions.offerOnCreate = false;
+    testOptions.clientTooSlowPolicy = iox::popo::ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
+
+    iox::popo::ServerOptions::deserialize(testOptions.serialize())
+        .and_then([&](auto& roundTripOptions) {
+            EXPECT_THAT(roundTripOptions.requestQueueCapacity, Ne(defaultOptions.requestQueueCapacity));
+            EXPECT_THAT(roundTripOptions.requestQueueCapacity, Eq(testOptions.requestQueueCapacity));
+
+            EXPECT_THAT(roundTripOptions.nodeName, Ne(defaultOptions.nodeName));
+            EXPECT_THAT(roundTripOptions.nodeName, Eq(testOptions.nodeName));
+
+            EXPECT_THAT(roundTripOptions.offerOnCreate, Ne(defaultOptions.offerOnCreate));
+            EXPECT_THAT(roundTripOptions.offerOnCreate, Eq(testOptions.offerOnCreate));
+
+            EXPECT_THAT(roundTripOptions.clientTooSlowPolicy, Ne(defaultOptions.clientTooSlowPolicy));
+            EXPECT_THAT(roundTripOptions.clientTooSlowPolicy, Eq(testOptions.clientTooSlowPolicy));
+        })
+        .or_else([&](auto&) { FAIL() << "Serialization/Deserialization of ServerOptions failed!"; });
+}
+
+TEST(ServerOptions_test, DeserializingBogusDataFails)
+{
+    const auto bogusSerialization = iox::cxx::Serialization::create("hypnotoad", "brain slug", "rock star");
+    iox::popo::ServerOptions::deserialize(bogusSerialization)
+        .and_then([&](auto&) { FAIL() << "Deserialization is expected to fail!"; })
+        .or_else([&](auto&) { SUCCEED(); });
+}
+
+using QueueFullPolicyUT = std::underlying_type_t<iox::popo::QueueFullPolicy2>;
```

Comment:
`QueueFullPolicyUT` doesn't seem to be used.

Reply:
You are right, C&P leftover